drivers/can/can.c

```diff
@@ -507,7 +507,7 @@ static ssize_t can_read(FAR struct file *filep, FAR char *buffer,
   FAR struct can_dev_s     *dev = inode->i_private;
 #endif
 
-  caninfo("buflen: %lu\n", buflen);
+  caninfo("buflen: %d\n", buflen);
```

Comment:
Not like this, use %zu or cast buflen to (unsigned long)

Reply:
Actually, use cast, not %zu. Casting to unsigned long keeps it C89

Reply:
casting to unsigned long